### PR TITLE
Create GH action for automated docker builds on releases

### PR DESCRIPTION
**Pull request by justusschock**
As we discussed, this is a first draft of publishing docker images on releases.

cc @williamFalcon @Borda

### CODE REVIEW

File: .github/workflows/docker_builds.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Publish Docker
+on: [push]
```
Comment (Borda) [inline .github/workflows/docker_builds.yml:2]: I would rather spec push to master
https://github.com/PyTorchLightning/pytorch-lightning/blob/4d24032ea5d28dca4b3a7aefece2fd28572f7b10/.github/workflows/pypi-release.yml#L7-L12

File: .github/workflows/docker_builds.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
```
Comment (Borda) [inline .github/workflows/docker_builds.yml:7]: rather spec a version, master can be bad some time...

**Comment by justusschock** [comment]
@Borda the docket image that will be created depends on the dockerfile we will provide here

**Comment by Borda** [comment]
> @Borda the docket image that will be created depends on the dockerfile we will provide here

just don't see the `Dockerfile` or command/path in the action...

**Comment by justusschock** [comment]
@Borda I added a build matrix as well as build arguments for the docker file and it's location along with your other requested changes. All paths and arguments for the docker file are based on #1569

File: .github/workflows/docker_builds.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    types: 
+       - created
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python_version: [3.6, 3.7, 3.8]
+        pytorch_version: [1.1, 1.2, 1.3, 1.4, 1.5]
```
Comment (Borda) [inline .github/workflows/docker_builds.yml:16]: Double check cuda version compatibility

**Comment by Borda** [comment]
@williamFalcon @justusschock I think it was not complete..

File: .github/workflows/docker_builds.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      # publish master
+      uses: elgohr/Publish-Docker-Github-Action@master
+      if: github.event_name == 'push'
+      with:
+        name: pytorchlightning/pytorch_lightning
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+        dockerfile: docker/Dockerfile
+        buildargs: PYTHON_VERSION=${{ matrix.python_version }},PYTORCH_VERSION=${{ matrix.pytorch_version }},LIGHTNING_VERSION=${{ steps.get_version.outputs.VERSION }}
+        tags: "latest_py${{ matrix.python_version }}_torch${{ matrix.pytorch_version }}"
```
[the remark this answers is not in the document]
Comment (justusschock) [reply .github/workflows/docker_builds.yml:44]: I would keep it. I think, that most people would like to install the latest stable version here (including myself) without the need to update the image manually.

File: .github/workflows/docker_builds.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        pytorch_version: [1.1, 1.2, 1.3, 1.4, 1.5]
+    steps:
+    - name: Extract Current Tag
+      if: contains(github.ref, 'refs/tags/')
+      id: get_version
+      run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+    - uses: actions/checkout@2.14
+    - name: Publish Releases to Docker
+      # only on releases
+      uses: elgohr/Publish-Docker-Github-Action@master
```
[the remark this answers is not in the document]
Comment (justusschock) [reply .github/workflows/docker_builds.yml:25]: I had a specific version. Seems like it broke during rebase. Will add that again

**Comment by justusschock** [comment]
will continue in #1613

File: .github/workflows/docker_builds.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/docker_builds.yml:44]: I mean you have master commit 234, then you add 5 more commits to the master and you create a release from the past commit 234...
If I read this correctly the Docker will be created from 234 + 5, not 234 right?

**Pull request merged**